codebundles/aws-c7n-ebs-health/sli.robot

```diff
@@ -0,0 +1,94 @@
+    ${unattached_ebs_event_score}=    Evaluate    1 if int(${count.stdout}) >= int(${EVENT_THRESHOLD}) else 0
+    Set Global Variable    ${unattached_ebs_event_score}
+
+Check Unencrypted EBS Volumes in `${AWS_REGION}`
+    [Documentation]  Check for unencrypted EBS volumes and report any found that do not meet encryption requirements.
+    [Tags]    ebs    storage    aws    security    volume
+    ${c7n_output}=    RW.CLI.Run Cli
+    ...    cmd=custodian run -r ${AWS_REGION} --output-dir ${OUTPUT_DIR}/aws-c7n-ebs-health ${CURDIR}/unencrypted-ebs-volumes.yaml --cache-period 0
+    ...    secret__aws_access_key_id=${AWS_ACCESS_KEY_ID}
+    ...    secret__aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}
+    ${count}=     RW.CLI.Run Cli
+    ...    cmd=cat ${OUTPUT_DIR}/aws-c7n-ebs-health/unencrypted-ebs-volumes/metadata.json | jq '.metrics[] | select(.MetricName == "ResourceCount") | .Value'
+    ${unencrypted_ebs_event_score}=    Evaluate    1 if int(${count.stdout}) >= int(${SECURITY_EVENT_THRESHOLD}) else 0
+    Set Global Variable    ${unencrypted_ebs_event_score}
+
+
+Check Unused EBS Snapshots in `${AWS_REGION}`
+    [Documentation]  Check for unused EBS snapshots. 
+    [Tags]    ebs    storage    aws    snapshots    volume
+    ${c7n_output}=    RW.CLI.Run Cli
+    ...    cmd=custodian run -r ${AWS_REGION} --output-dir ${OUTPUT_DIR}/aws-c7n-ebs-health ${CURDIR}/unused-ebs-snapshots.yaml --cache-period 0
+    ...    secret__aws_access_key_id=${AWS_ACCESS_KEY_ID}
+    ...    secret__aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}
+    ${count}=     RW.CLI.Run Cli
+    ...    cmd=cat ${OUTPUT_DIR}/aws-c7n-ebs-health/unused-ebs-snapshots/metadata.json | jq '.metrics[] | select(.MetricName == "ResourceCount") | .Value'
+    ${unsued_ebs_snapshot_event_score}=    Evaluate    1 if int(${count.stdout}) >= int(${EVENT_THRESHOLD}) else 0
+    Set Global Variable    ${unsued_ebs_snapshot_event_score}
+
+
+Generate EBS Score
+    ${ebs_health_score}=      Evaluate  (${unattached_ebs_event_score} + ${unencrypted_ebs_event_score} + ${unsued_ebs_snapshot_event_score}) / 3
+    ${health_score}=      Convert to Number    ${ebs_health_score}  2
+    RW.Core.Push Metric    ${health_score}
+
+** Keywords ***
+Suite Initialization
+    ${AWS_REGION}=    RW.Core.Import User Variable    AWS_REGION
+    ...    type=string
+    ...    description=AWS Region
+    ...    pattern=\w*
+    ${AWS_ACCOUNT_ID}=    RW.Core.Import User Variable   AWS_ACCOUNT_ID
+    ...    type=string
+    ...    description=AWS Account ID
+    ...    pattern=\w*
+    ${AWS_ACCESS_KEY_ID}=    RW.Core.Import Secret   AWS_ACCESS_KEY_ID
+    ...    type=string
+    ...    description=AWS Access Key ID
+    ...    pattern=\w*
+    ${AWS_SECRET_ACCESS_KEY}=    RW.Core.Import Secret   AWS_SECRET_ACCESS_KEY
+    ...    type=string
+    ...    description=AWS Access Key Secret
+    ...    pattern=\w*
+    ${EVENT_THRESHOLD}=    RW.Core.Import User Variable    EVENT_THRESHOLD
```

Comment:
Defaults should be 0. 

Consider adding this to the runbook as well, and then generating issues if the length of the list (of unattached / unused / unencrypted) volumes is > than the event threshold

```
    ${EVENT_THRESHOLD}=    RW.Core.Import User Variable    EVENT_THRESHOLD
    ...    type=string
    ...    description=The minimum number of EBS volumes | snapshots to consider unhealthy.
    ...    pattern=^\d+$
    ...    example=2
    ...    default=1
    ${SECURITY_EVENT_THRESHOLD}=    RW.Core.Import User Variable    SECURITY_EVENT_THRESHOLD
    ...    type=string
    ...    description=The minimum number of security-related EBS volumes to consider unhealthy.
    ...    pattern=^\d+$
    ...    example=2
    ...    default=1
```

Reply:
Updated `Defaults` to 0.

I assume this gives us the number of items found in each task.

`${unsued_ebs_snapshot_event_score}=    Evaluate    1 if int(${count.stdout}) > int(${EVENT_THRESHOLD}) else 0`

and then evaluate like this:

`${ebs_health_score}=      Evaluate  (${unattached_ebs_event_score} + ${unencrypted_ebs_event_score} + ${unsued_ebs_snapshot_event_score}) / 3`

Let me know what can be improved here.

Reply:
`${unsued_ebs_snapshot_event_score}= Evaluate 1 if int(${count.stdout}) > int(${EVENT_THRESHOLD}) else 0` should be `<` and not `>`. See that 1 = good, 0 = bad. If the count is less than the threshold, it's "good"

Reply:
There is a case if user go with default THRESHOLD = 0 and count's lowest value is 0 so if `count.stdout` is 0 then it will still be bad!